src/main/java/com/feedzai/commons/sql/abstraction/engine/AbstractDatabaseEngine.java

```diff
@@ -398,20 +398,12 @@ public synchronized void close() {
                         } catch (SQLException ex) {
                             logger.debug(String.format("Failed to flush before dropping entity '%s'", me.getValue().getEntity().getName()), ex);
                         } finally {
-                            if (insert != null) {
-                                try {
-                                    insert.close();
-                                } catch (Exception e) {
-                                    logger.trace("Could not close prepared statement.", e);
-                                }
-                            }
 
-                            if (insertReturning != null) {
-                                try {
-                                    insertReturning.close();
-                                } catch (Exception e) {
-                                    logger.trace("Could not close prepared statement.", e);
-                                }
+                            try {
+                                me.getValue().close();
+                            } catch (Exception e) {
+                                // Should never be thrown
+                                logger.trace("Could not close insert statements.", e);
```

Comment:
This is an erroneous situation, but from which we are going to gracefully exit, so let's use WARN logging instead. Also, maybe refer that the insert statements are for a "MappedEntity" (i.e, they are not "custom" prepared statements as the one below)

Reply:
Addressed in b449999